packages/server/src/api/com/atproto/account.ts

```diff
@@ -81,63 +62,111 @@ export default function (server: Server) {
       isTestUser = true
     }
 
-    // verify username and email are available.
+    const { did } = await db.transaction(async (dbTxn) => {
+      if (config.inviteRequired) {
+        if (!inviteCode) {
+          throw new InvalidRequestError(
+            'No invite code provided',
+            'InvalidInviteCode',
+          )
+        }
 
-    // @TODO consider pushing this to the db, and checking for a
-    // uniqueness error during registerUser(). Main blocker to doing
-    // that now is that we need to create a did prior to registration.
+        await dbTxn.lockTable('invite_code_use')
```

Comment:
not thrilled with throwing a manual lock on the table... it's better than using SERIALIZABLE txs, and I couldn't find another way of doing this. lmk if you ahve any other ideas

Reply:
Scratch that.  Selecting the invite code `FOR UPDATE` (even though it's not going to be updated) is simpler than both of those options, and gives you the per-invite granularity 🙃

Reply:
Here are some options for this: https://github.com/bluesky-social/atproto/pull/253